### PR TITLE
Issue #7256: Rename viewer-role to reader-role.

### LINKED ISSUE #7256: Rename viewer-role to reader-role
> This work depends on [WS-CD-Open 15384](https://github.ibm.com/was-liberty/WS-CD-Open/issues/15384) being completed and in master. Failure to do so will result in failed FATs.

The 'viewer-role' element needs to be renamed 'reader-role'. Additionally, search for any reference to 'viewer' in the context of an authorization role and replace it with 'reader'.

### PR DESCRIPTION
**Pull request by jvanhill**
fixes #7256

### CODE REVIEW

**Comment by jvanhill** [comment]
Build with WS-CD-Open changes:

https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_HdBWpmJBEemP6vsjGLPPfg

File: dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -18,8 +18,8 @@
 role.administrator=Administrator Role
 role.administrator.desc=A collection of users, groups, or both that are assigned the server administrator role.
 
-role.viewer=Viewer Role
-role.viewer.desc=A collection of users, groups, or both that are assigned the server viewer role.
+role.reader=Reader Role
```
Comment (dmuelle) [inline dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties]: ok

File: dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -18,8 +18,8 @@
...
+role.reader.desc=A collection of users, groups, or both that are assigned the server reader role.
```
Comment (dmuelle) [inline dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties]: All good on the change from Viewer to Reader- but- should this description include what what the Reader role entails? (eg. capable of viewing but editing) Is Reader a standard role with standard permissions that could be succinctly described? As is, this description doesn't describe the role, just restates what's implied by the title

Comment (jvanhill) [reply dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties]: Yeah. I questioned this myself. 

Maybe we could do something like:

> reader.desc=A collection of users, groups, or both that are assigned the server reader role. The reader role bestows read-only privileges.

Comment (jvanhill) [reply dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties]: Could update the the administrator description in the same fashion, but saying 'bestows reader and write privileges.'

Comment (jvanhill) [reply dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties]: Maybe some ideas from here?

https://www.ibm.com/support/knowledgecenter/ro/SSAW57_9.0.0/com.ibm.websphere.nd.multiplatform.doc/ae/rsec_adminroles.html

**Comment by jvanhill** [comment]
Build with WS-CD-Open changes:

https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_8zmgcGK6EemP6vsjGLPPfg

File: dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties (hunk 2 above)
Comment (dmuelle) [reply dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties]: I think that is an improvement- what about "grants" instead of "bestows"?

Also, "server reader role" is a bit hard to parse at a glance. Would "...Reader Role for the server" also be correct?

I notice that, across KC, Reader Role is capitalized. I can see why it might not be in this description, but I think it could help with legibility and consistency to capitalize it, unless there is a specific reason not to.

**Comment by jvanhill** [comment]
The last build appeared to fail b/c of a network issue contacting a build server. Respinning.

https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_oeZzoGUEEemEm9vQdIVPyw

**Comment by skasund** [comment]
No message review from L2 required.

File: dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties (hunk 2 above)
Comment (jvanhill) [reply dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties]: Does this work?

role.reader=Reader Role
role.reader.desc=A collection of users, groups, or both that are granted the Reader role.

Comment (dmuelle) [reply dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties]: That works- but I think you include permission description you had above, so:

"role.reader=Reader Role
role.reader.desc=A collection of users, groups, or both that are assigned the Reader role. The reader role grants read-only privileges."

Comment (jvanhill) [reply dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties]: I have updated both the reader and administrator role descriptions accordingly.

**Comment by jvanhill** [comment]
Rebuild with changes to message with WS-CD-Open changes:

https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_6SS70GagEemEm9vQdIVPyw

File: dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties (hunk 2 above)
Comment (jvanhill) [reply dev/com.ibm.ws.management.security/resources/OSGI-INF/l10n/metatype.properties]: David approved the changes.

**Comment by jvanhill** [comment]
Infrastructure build issues. Respinning.

https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_m6JAwGqEEemEm9vQdIVPyw

**Comment by jvanhill** [comment]
Infrastructure build issues. Respinning.

https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_EetdkG3VEemEm9vQdIVPyw

**Comment by jvanhill** [comment]
respinning due to infrastructure build issues:

https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewDefinition&id=_IQRVkGV1EeaugeK15nw0Xw

**Comment by jvanhill** [comment]
#build

**Comment by jvanhill** [comment]
Had one test failure that appears to be due to the changes. Had to update the descriptions of the roles. Respinning the build.

#build

**Comment by jvanhill** [comment]
The last build failed due to a scheduled RTC outage. Respinning.

#build

**Comment by jvanhill** [comment]
None of the test failures appear to be related to my changes.
